### PR TITLE
test: algo to function in substratools

### PR DESCRIPTION
**Pull request by ThibaultFy**
Signed-off-by: ThibaultFy <50656860+ThibaultFy@users.noreply.github.com>

Companion PR:

- MAIN: https://github.com/Substra/substra-tools/pull/63
- https://github.com/Substra/substra/pull/303
- https://github.com/Substra/substrafl/pull/34
- https://github.com/Substra/substra-documentation/pull/201
- https://github.com/Substra/substra-tests/pull/224

## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [x] If necessary, the [changelog](https://github.com/Substra/substra/blob/main/CHANGELOG.md) has been updated
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification
- For any breaking changes, companion PRs have been opened on the following repositories:
  - [x] [substra-tests](https://github.com/Substra/substra)
  - [x] [substrafl](https://github.com/Substra/substrafl)
  - [x] [substra-documentation](https://github.com/Substra/substra-documentation)

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def test_command_config(workdir):
     assert list(cfg.keys()) == expected_profiles
 
 
-def mock_client_call(mocker, method_name, response="", side_effect=None):
-    return mocker.patch(f"substra.cli.interface.Client.{method_name}", return_value=response, side_effect=side_effect)
+def mock_client_call(mocker, function_name, response="", side_effect=None):
```
Comment (Fabien-GELUS) [inline tests/test_cli.py:76]: This one is not related to substratools

**Review by Fabien-GELUS — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,7 +102,7 @@ compute_plan = client.add_compute_plan(
 - used openers within the library only exposes `get_data` and `fake_data` methods due to breaking changes within
   substra-tools
 - a third argument `task_properties` (containing the `rank` of a task) has been added to all algo methods relying on substra-tools
-- substra tools metrics `tools.Metrics` become `tools.MetricAlgo` and `tools.metrics.execute` becomes `tools.algo.execute` (#290)
+- substra tools metrics `tools.Metrics` become `tools.MetricAlgo` and `tools.metrics.execute` becomes `tools.execute` (#290)
```
Comment (Esadruhn) [inline CHANGELOG.md:105]: do not change the previous changelog entries

File: CHANGELOG.md [hunk 3]
```diff
@@ -535,5 +539,4 @@ client1_org_id = clients[0].organization_info().organization_id
 
 ## [0.10.0](https://github.com/Substra/substra/releases/tag/0.10.0) - 2021-08-05
 
-[0.10.0]: https://github.com/Substra/substra/compare/0.9.0...0.10.0
```
Comment (Esadruhn) [inline CHANGELOG.md:538]: why change this?

**Review by Esadruhn — APPROVED** [review]
